src/compiler/checker.ts

```diff
@@ -4979,16 +4983,26 @@ namespace ts {
             return checkTypeRelatedTo(source, target, assignableRelation, errorNode, headMessage, containingMessageChain);
         }
 
+        function isSignatureAssignableTo(source: Signature,
+                                         target: Signature,
+                                         ignoreReturnTypes: boolean): boolean {
+            return compareSignaturesRelated(source, target, ignoreReturnTypes, /*errorReporter*/ undefined, compareTypesAssignable) !== Ternary.False;

```

Comment:
I think you could just pass isTypeAssignableTo here, instead of having compareTypesAssignable? Is your concern about the boolean versus Ternary return type? I think the callback can just return `boolean | Ternary`.